src/snowflake/snowpark/_internal/data_source/datasource_partitioner.py

```diff
@@ -87,6 +89,13 @@ def __init__(
             else None
         )
 
+    @classmethod
+    def from_dict(cls, init_dict: dict) -> "DataSourcePartitioner":
+        """Create safely from a dict, ignoring extra fields."""
+        params = inspect.signature(cls).parameters
+        filtered = {k: v for k, v in init_dict.items() if k in params}
```

Comment:
if we switch to the normal dict approach instead of locals, you probably won't need the filter logic